### ISSUE TITLE
Preferred file extension?

### ISSUE THREAD

**Issue by giuliano-oliveira** [msgpack/msgpack#291]
I agree what @frsyuki  said [here](https://github.com/msgpack/msgpack/pull/267#issuecomment-591787926)

> MessagePack is not intended to be a file format but intended to be a generic object serialization format, which means that we have a lot more room to design file format on top of msgpack.

However i am using it as a file format in my python app since i didn't want to use pickle as it is insecure and wanted a binary serializer, so, i would like to know what file extension do you guys use in this format?

**Comment by markusschaber** [comment]
As this will be a format specific to your application, you should find an extension which matches your application.

**Comment by maisiliym** [comment]
`.msgpack` Is most sensible, for clarity, and because whatever system is reading file paths does not care about shaving 3-4 bytes off it.

**Comment by dukelec** [comment]
Short extension: `.mpk` (preferred)
Full extension: `.msgpack` (optional)

**Comment by allan-bonadio** [comment]
You should add the above to your spec.  I was looking for it, and it was nowhere to be found.

I agree with the idea that each specific use should have its own filename suffix.  It drives me crazy that there's so many different kinds of xml and json files and they all have the same stupid .xml or .json suffix on the filename.  We don't take our .html and .c and .js files and give them a '.txt' suffix!

But, we do use a .txt suffix, sometimes.  For files that just don't have any particular format, files that are unique, and files we just can't think up a better suffix for.  README files that aren't .md .

**Comment by analog-cbarber** [comment]
The issue is that .c and .js are special cases of text files. They are also widespread public standards so anyone who sees those extensions will be able to find a suitable program to read those files. This may not true for your own special program. In fact, this might just hold random data used in research or the like. This is why it is nice to have a format-specific file extension. If I see a file with a `.json` extension I might not know what the actual data is, but at least I have a reasonable expectation of using standard JSON apis to look at it. This is even more important for binary formats.

**Comment by allan-bonadio** [comment]
Agreed, .mpk as a suffix is too general for most purposes.  But I visualize myself at the keyboard, designing software as I'm writing it.  I need a filename for a msgpack file.  I'm not even sure of exactly how it'll be used, but I can change the name later.  Maybe it's just a hack I'll abandon in a few hours or days.  That's when I'd want to use a .mpk suffix.  And, we'd all call it 'the messagepack file' for a while.  and, I'd be the one to insist we give it a real, purpose-related name and suffix.  But that's me, and I know most programmers don't bother.

But, I get what you're saying, the risk is that people will use a .mpk suffix for every msgpack file.  Maybe if you **vehemently recommend** that they dream up their own suffix?  I shouldn't tell you how to run your project.

**Comment by mlsomers** [comment]
The nice thing about standard suffix is that you can instruct your operating system to open it in a specific application when someone double-clicks it. For MsgPack files I use my MsgPackExplorer.exe application ([screenshots here](http://www.infotopie.nl/open-source/msgpack-explorer), can be [downloaded here](https://github.com/mlsomers/LsMsgPack/releases)). It saves you the hassle of first finding and running the application and then navigating to the file using the "open file" dialog box in the application...

If you live in commandline shell and hate GUI's then you might not appreciate these benefits to the same extent, but I still think that naming different xml files with the same xml suffix makes allot of sense. You instantly know what [specific editor](https://www.xmlfox.com/) to use in order to inspect them.

It's a long time ago that I would default to a hex-editor to peek, then exit and start a more appropriate app. For MsgPack I don't really think it would directly be obvious to an average user that they are looking MsgPack when opening in a hex-editor (unless doing it frequently and building muscle-memory knowing what to look for).

Having a specific suffix also means having an explicit keyword to search for information on the file type.
